src/target/riscv/riscv-013.c

```diff
@@ -977,27 +977,51 @@ static int execute_abstract_command(struct target *target, uint32_t command,
 	return ERROR_OK;
 }
 
-static riscv_reg_t read_abstract_arg(struct target *target, unsigned index,
-		unsigned size_bits)
+static void add_data_regs_read(struct riscv_batch *batch, unsigned int index,
+		unsigned int size_bits)
 {
+	const unsigned int size_in_words = size_bits / 32;
+	assert(size_bits % 32 == 0);
+	const unsigned int offset = index * size_in_words;
+	for (unsigned int i = 0; i < size_in_words; ++i) {
+		const unsigned int reg_address = DM_DATA0 + offset + i;
+		riscv_batch_add_dm_read(batch, reg_address);
+	}
+}
+
+static riscv_reg_t read_data_regs(struct riscv_batch *batch,
```

Comment:
(similar to previous)

```suggestion
static riscv_reg_t abstract_data_get_from_batch(struct riscv_batch *batch,
```

Reply:
Addressed.